File: CPlusPlus/Algorithms/DFS.cpp
#include <list>

// Example class of a graph implementation:
class Graph {
    public:
        Graph::Graph(int Size) 
            :Size(Size) {
            adj = new std::list<int>[Size];
        }
        std::list<int>* adj;
        int Size;
};


void dfs(Graph & G, const int startVert) {

    bool* visited = new bool[G.Size];
    for(int i = 0; i < G.Size; ++i) {
        visited[i] = false;
    }

    std::list<int> queue;

    // Mark and enqueue our start vert
    visited[startVert] = true;
    queue.push_front(startVert);

    while(!queue.empty()) {
        // Get the next vert
        int v = queue.front();
        queue.pop_front();
        // Explore it:
        std::list<int>::iterator it;
        for(it = G.adj[v].begin(); it != G.adj[v].end(); ++it) {

            if(!visited[*it]) {
                visited[*it] = true;
                queue.push_front(*it);
            }

        }
    }
    delete[] visited;
}

void dfsAux(Graph & G, int v, bool* visited) {
    visited[v] = true;

    std::list<int>::iterator it;
    for(it = G.adj[v].begin(); it != G.adj[v].end(); ++it) {

        if(!visited[*it]) {
            // pre(); something can happen before the visit.
            dfsAux(G, *it, visited);
            // post(); something can happpen after the visit.
        }
    }
}

void dfsRecursive(Graph & G, const int startVert) {
    bool* visited = new bool[G.Size];
    dfsAux(G, startVert, visited);
    delete[] visited;
}

File: PHP/Algorithms/DFS.php
<?php

class DFS {

    public static function dfsIterative($G, $startVert) {
        $visited = array();
        for($i = 0; $i < $G->size; $i++) {
            array_push($visited, false);
        }
        $s = SplStack();
        
        $s->push($startVert);

        while(!$s->isEmpty()) {
            $v = $s->pop();
            foreach($G->adj[$v] as $adjV) {
                if(!$visited[$adjV]) {
                    $visited[$adjV] = true;
                    $s->push($adjV);
                }
            }
        }
    }

    public static function dfsRecursive($G, $startVert) {
        $visited = array();
        for($i = 0; $i < $G->size; $i++) {
            array_push($visited, false);
        }
        DFS::dfsAux($G, $startVert, $visited);
    }

    public static function dfsAux($G, $v, $visited) {
        $visited[$v] = true;
        foreach($G->adj[$v] as $adjV) {
            if(!$visited[$adjV]) {
                DFS::dfsAux($G, $adjV, $visited);
            }
        }
    }
}

// Example class of what a graph might look like.

class Graph {
    public $adj;
    public $size;
}